src/runtime_src/core/common/ishim.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef core_common_ishim_h
+#define core_common_ishim_h
+
+#include "xrt.h"
+#include "error.h"
+#include <stdexcept>
+
+namespace xrt_core {
+
+/**
+ * struct ishim - Shim API implemented by core libraries
+ *
+ * All methods throw on error
+ */
+struct ishim
```

Comment:
Hmm...since the majority of methods in the ishim and shim class are helper methods and they don't seem to change the class variables, shouldn't they be "const" or would that make them too restrictive?